src/OpenLoco/src/World/CompanyAi/CompanyAi.cpp

```diff
@@ -2163,12 +2163,411 @@ namespace OpenLoco
+    static void sub_48137F(AiThought& thought)
+    {
+        thought.numStations = kThoughtTypeNumStations[enumValue(thought.type)];
+        for (auto i = 0U; i < thought.numStations; ++i)
+        {
+            auto& aiStation = thought.stations[i];
+            aiStation.var_02 = AiThoughtStationFlags::none;
+        }
+
+        if (thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk11))
+        {
+            thought.var_04 = 1;
+        }
+        else if (thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk6))
+        {
+            if (getCurrentYear() < 1945)
+            {
+                thought.var_04 = 5;
+            }
+            else if (getCurrentYear() < 1991)
+            {
+                thought.var_04 = 6;
+            }
+            else
+            {
+                thought.var_04 = 7;
+            }
+        }
+        else
+        {
+            const auto distance = distanceBetweenDestinations(thought);
+            auto baseStationLength = (std::max(0, distance - 32 * 28) / 1024) + 5;
+            auto yearAdditionalLength = 0;
+            if (getCurrentYear() >= 1925 && getCurrentYear() < 1955)
+            {
+                yearAdditionalLength = 1;
+            }
+            else if (getCurrentYear() < 1985)
+            {
+                yearAdditionalLength = 2;
+            }
+            else
+            {
+                yearAdditionalLength = 3;
+            }
+            const auto minLength = thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::unk17) ? 7 : 0;
+            thought.var_04 = std::clamp(baseStationLength + yearAdditionalLength, minLength, 11);
+        }
+    }
+
+    // 0x00481D6F
+    static bool sub_481D6F(const World::Pos2& pos)
+    {
+        auto tilePosA = toTileSpace(pos) - TilePos2{ 2, 2 };
+        auto tilePosB = toTileSpace(pos) + TilePos2{ 2, 2 };
+        auto numBuildings = 0U;
+        for (const auto& tilePos : TilePosRangeView(tilePosA, tilePosB))
+        {
+            auto tile = World::TileManager::get(tilePos);
+            for (auto& el : tile)
+            {
+                auto* elBuilding = el.as<World::BuildingElement>();
+                if (elBuilding != nullptr)
+                {
+                    numBuildings++;
+                }
+            }
+        }
+        return numBuildings < 4;
+    }
+
+    // 0x004814D6
+    static bool sub_4814D6(AiThought& thought)
+    {
+        auto destinationPosition = [](bool isIndustry, uint8_t destination) {
+            if (isIndustry)
+            {
+                const auto* industry = IndustryManager::get(static_cast<IndustryId>(destination));
+                return Pos2{ industry->x, industry->y };
+            }
+            else
+            {
+                auto* town = TownManager::get(static_cast<TownId>(destination));
+                return Pos2{ town->x, town->y };
+            }
+        };
+        if (thoughtTypeHasFlags(thought.type, ThoughtTypeFlags::singleDestination))
```

Comment:
This is a really long if/else block, with the if part also containing many branches. Even if we don't tackle the inner ones, I think we should try splitting the outer two into a function dealing with single destination case, and one for the other.

Reply:
Okay think that has been addressed.